nemo/lightning/pytorch/trainer.py

```diff
@@ -1,15 +1,29 @@
 from copy import deepcopy
+from typing import List, Optional, Union
 
 import fiddle as fdl
 import pytorch_lightning as pl
+from pytorch_lightning.callbacks import Callback
 from typing_extensions import Self
 
 from nemo.lightning.fabric.conversion import to_fabric
 from nemo.lightning.fabric.fabric import Fabric
 from nemo.lightning.io.mixin import IOMixin, serialization, track_io
+from nemo.lightning.pytorch.callbacks import MegatronSetup
 
 
 class Trainer(pl.Trainer, IOMixin):
+    def __init__(self, callbacks: Optional[Union[List[Callback], Callback]] = None, **kwargs):
```

Comment:
Please remove this, we don't want to overwrite the `__init__` here since we don't want to loose auto-complete. You can add the callback to the Trainer inside the setup inside the `FSDPStrategy`.